### ISSUE TITLE
[BUG]Paplay playback has repetitive overrun and drained after waking up from s3 during paplay.

### ISSUE THREAD

**Issue by sinahuang** [thesofproject/linux#1411]
**Describe the bug**
Paplay playback stream drained when wake up from s3 during paplay.

**To Reproduce**
1.Paplay speaker/HDMI/DP via command "paplay wav/paplay wav/48California_Dreaming.wav -vvv"
2.Do s3 via "sudo rtcwake -m mem -s 10"

**Frequency**
4/5

**Expected result**
Paplay output normally after wake up from s3

**Actual result**
Paplay stopped and playback stream drained after wake up from s3.

Recover method: Trigger Paplay again can recover and output normally.
```
ubuntu@sh-icl-rvp-hda-001:~$ paplay wav/48California_Dreaming.wav -vvv
Opening a playback stream with sample specification 's16le 2ch 48000Hz' and channel map 'front-left,front-right'.
Connection established.
Stream successfully created.
Buffer metrics: maxlength=4194304, tlength=384000, prebuf=380164, minreq=3840
Using sample spec 's16le 2ch 48000Hz', channel map 'front-left,front-right'.
Connected to device alsa_output.pci-0000_00_1f.3-platform-sdw_rt711_rt1308_rt715.HiFi__hw_sofsdwrt7111308_6__sink (index: 2, suspended: no).
Stream started.
Stream underrun.
Stream started.
Stream underrun.
Stream started.
Stream underrun.
Stream started.
Stream underrun.
Stream started.
Stream underrun.
Stream started.
Stream underrun.
Stream started.
Playback stream drained.y: 68829416 usec.
Draining connection to server.

```

Platform: CML Chrome I2S
Firmware: 59c9338 https://github.com/thesofproject/sof/commits/master
Kernel: f5e70ce https://github.com/thesofproject/linux/commits/topic/sof-dev
Topology: file: tools/topology/sof-cml-rt5682-max98357a.tplg same as firmware
Pulseaudio:11.1

[dmesg.log](https://github.com/thesofproject/linux/files/3792785/dmesg.log)
[logger.log](https://github.com/thesofproject/linux/files/3792786/logger.log)

**Comment by Liviali155** [comment]
on APL UP2 I2S PCM512x,
Paplay Headset  via command "paplay  48California_Dreaming.wav -vvv"
reproduced this issue

**Comment by Liviali155** [comment]
1.issue can be reproduced on:
    CML laptop  with onboard ALC3204 in HDA mode  (pulseaudio 11.1)
    APL UP2 with nocodec   (  pulseaudio 11.1)                                    
    APL UP2 with codec PCM512X in I2S mode    ( pulseaudio 11.1)
    TGL RVP with onboard RT711 in SDW mode   (pulseaudio 11.1)
    CML Laptop with onboard RT711+1308+715 in SDW mode  (pulseaudio 11.1)
    CML Laptop with onboard ALC3204 in Legacy HDA mode (pulseaudio 11.1)

2.issue can't reproduce on:
   CFL-S RVP with  onboard ALC700 in HDA mode  (pulseaudio 11.1)
   ICL RVP with  onboard ALC700 in HDA mode (pulseaudio 11.1)

3.Platforms blocked with #1387  :
   GLK Chrome with  onboard DA7219 in I2S mode  (pulseaudio 11.1)
  CML Chrome with  onboard RT5682 in I2S mode    (pulseaudio 11.1)

**Comment by ClarexZhou** [comment]
Error not occurred on release/sof-v5.0 (abbc71ca) with pulseaudio 11.1, might related with #1000 
Stream underrun is also display when wake up. But paplay isn't stop, still continues to play.
**Test Recipe**	
Kernel repo	https://github.com/thesofproject/linux/commits/release/sof-v5.0 (abbc71ca) 
Firmware repo	https://github.com/thesofproject/sof/commits/cml-hda-dmic-002-drop-stable (0f736283)
Topology: 	Same as above SOF repo, sof-hda-generic-2ch.tplg rename to sof-hda-generic.tplg
Jaska UCM	https://github.com/juimonen/alsa-ucm-conf-1/tree/dell_test (3333df89)
Platform: CML laptop with ALC3204 in HDA mode

**Comment by wenqingfu** [comment]
Since #1000 is quoted and it is PA related, please amend the bug description with PulseAudio version used in testing.

**Comment by ClarexZhou** [comment]
> Since #1000 is quoted and it is PA related, please amend the bug description with PulseAudio version used in testing.

All platforms are using PA 11.1, no matter the issue reproduced and issue not reproduced ones. Seems not PA issue.

**Comment by dengyangchao** [comment]
Tried on CFL notebook with onboard codec ALC3254 in HDA mode:
1. Ubuntu18.04 & pulseaudio11.1  
use paplay command or play through Browser both can reproduce the issue.
2. Ubuntu19.04 & pulseaudio12.2
use paplay command or play through Browser both can reproduce the issue.

**Environment**
Firmware: 63a55299 https://github.com/thesofproject/sof/commits/master
Kernel: ecc776ae https://github.com/thesofproject/linux/commits/topic/sof-dev

**Comment by juimonen** [comment]
ack, I'll check this.

**Comment by juimonen** [comment]
but isn't this exactly same as https://github.com/thesofproject/linux/issues/1000?

@kv2019i is exactly describing this use case and his patch to pulseaudio fixes this (I tested it and it works). And it is fixed in post V11.1 pulseaudio's....

So would says this is Pulseaudio issue.

**Comment by kv2019i** [comment]
@juimonen ack, if the patch for #1000 helps, then this is indeed a bug in pulseaudio with a known fix available

**Comment by dengyangchao** [comment]
@juimonen @kv2019i  But if we keep pulseaudio 11.1 and change kernel from ```sof-dev``` to ```release/sof-v5.0``` issue can't reproduce. Even in ```sof-dev``` with pulseaudio 11.1, some platforms also can't reproduce the issue. 
Details can see @Liviali155 @ClarexZhou comments.

**Comment by mengdonglin** [comment]
@dengyangchao Based on the comment https://github.com/thesofproject/linux/issues/1000#issuecomment-508128140 , PA12.2 still has the bug.  Please verify the PA patch for #1000 or latest upstream PA. Maybe Ubuntu 19.10?

 https://github.com/thesofproject/linux/tree/release/sof-v5.0 may hide the PA issue so cannot reproduce this issue.

**Comment by kv2019i** [comment]
@mengdonglin This is fixed 19.10 that has PA13:

http://releases.ubuntu.com/19.10/
https://packages.ubuntu.com/search?suite=eoan&searchon=names&keywords=pulseaudio

It is not clear why sof-v5.0 can avoid the bug, but it is likely just a question of timing. Based on earlier analysis, the bug clearly is on Pulseaudio side, so the robust way to fix this is on PA side.

**Comment by Liviali155** [comment]
Error not seen used ubuntu 19.10 with pulseaudio 13.0
platform: CFL notebook  with ALC3254  codec in HDA mode
Firmware: a62f5623https://github.com/thesofproject/sof/commits/master
Kernel: 0bdbe73a https://github.com/thesofproject/linux/commits/topic/sof-dev
Topology file: tools/topology/sof-hda-generic-2ch.tplg same as firmware
closed it
